app/bot/when.go

```diff
@@ -0,0 +1,134 @@
+package bot
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+)
+
+// When bot is answer on question "when the stream is started".
+type When struct {
+	now     func() time.Time
+	reactOn []string
+}
+
+// NewWhen makes a new When bot.
+func NewWhen(reactOn []string, nowFunc func() time.Time) *When {
+	log.Printf("[INFO] new when bot is started")
+
+	return &When{
+		reactOn: reactOn,
+		now:     nowFunc,
+	}
+}
+
+// Help returns help message
+func (w *When) Help() string {
+	return genHelpMsg(w.ReactOn(), "расписание эфиров Радио-Т")
+}
+
+// OnMessage returns one entry
+func (w *When) OnMessage(msg Message) Response {
+	if !contains(w.ReactOn(), msg.Text) {
+		return Response{}
+	}
+
+	return Response{
+		Text: when(w.now()),
+		Send: true,
+	}
+}
+
+// ReactOn keys
+func (w *When) ReactOn() []string {
+	return w.reactOn
+}
+
+func when(now time.Time) string {
+	const avgDuration = 2 * time.Hour
+	const whenPrefix = "[каждую субботу, 20:00 UTC](https://radio-t.com/online/)"
+	const tooSmallDuration = "пару секунд"
+
+	now = now.UTC()
+	prevStream, nextStream := closestPrevNextWeekdays(now, time.Saturday, 20, 0)
+	diffToPrev := -prevStream.Sub(now)
+	diffToNext := nextStream.Sub(now)
+
+	var whenCountdown string
+	switch {
+	case diffToPrev < avgDuration:
```

Comment:
to me, adding the third case as default will make it easier to understand

Reply:
I think I found the clear solution. Check this [commit](https://github.com/radio-t/super-bot/pull/83/commits/774804e16794072ccb60e91c3ee501c843a41c13)

Reply:
generally, I don't like if with else, and rather would use an early return in this case. But this is a nitpick, so whatever you prefer here is fine with me